components/admin_console/billing/company_info_display.tsx

```diff
@@ -59,51 +59,52 @@ const noCompanyInfoSection = (
-                        <FormattedMarkdownMessage
-                            id='admin.billing.company_info.employees'
-                            defaultMessage='{employees} employees'
-                            values={{employees: companyInfo.num_employees}}
-                        />
-                    </div>
-                    }
-                    <div className='CompanyInfoDisplay__companyInfo-addressTitle'>
-                        <FormattedMessage
-                            id='admin.billing.company_info.companyAddress'
-                            defaultMessage='Company Address'
-                        />
-                    </div>
-                    <div className='CompanyInfoDisplay__companyInfo-address'>
-                        <div>{address.line1}</div>
-                        {address.line2 && <div>{address.line2}</div>}
-                        <div>{`${address.city}, ${address.state}, ${address.postal_code}`}</div>
-                        <div>{address.country}</div>
-                    </div>
+    let body = noCompanyInfoSection;
+    const address = companyInfo.company_address.line1 ? companyInfo.company_address : companyInfo.billing_address;
+    body = (
+        <div className='CompanyInfoDisplay__companyInfo'>
+            <div className='CompanyInfoDisplay__companyInfo-text'>
+                <div className='CompanyInfoDisplay__companyInfo-name'>
+                    {companyInfo.name}
                 </div>
                 {supportAddCompanyInfo &&
-                <div className='CompanyInfoDisplay__companyInfo-edit'>
-                    <BlockableLink
-                        to='/admin_console/billing/company_info_edit'
-                        className='CompanyInfoDisplay__companyInfo-editButton'
-                    >
-                        <i className='icon icon-pencil-outline'/>
-                    </BlockableLink>
+                <div className='CompanyInfoDisplay__companyInfo-numEmployees'>
+                    <FormattedMarkdownMessage
+                        id='admin.billing.company_info.employees'
+                        defaultMessage='{employees} employees'
+                        values={{employees: companyInfo.num_employees}}
+                    />
                 </div>
                 }
+                <div className='CompanyInfoDisplay__companyInfo-addressTitle'>
+                    <FormattedMessage
+                        id='admin.billing.company_info.companyAddress'
+                        defaultMessage='Company Address'
+                    />
+                </div>
+                <div className='CompanyInfoDisplay__companyInfo-address'>
+                    <div>{address.line1}</div>
+                    {address.line2 && <div>{address.line2}</div>}
+                    <div>{`${address.city}, ${address.state}, ${address.postal_code}`}</div>
+                    <div>{address.country}</div>
+                </div>
```

Comment:
Should we include a case here if this address is empty to show something instead of empty commas? "Not available"?

Reply:
Yes, as soon as they sign up they only have Company Name filled in (it's part of the workspace creation) but they don't have anything else. They only will have the billing address once they upgrade. So they do have a company information with name but no address